app/frontend/actions.ts

```diff
@@ -982,7 +1000,129 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
   }
 
   const selectAdaliteStakepool = (state: State) => {
-    updateStakePoolIdentifier(getState(), null, state.poolRecommendation.recommendedPoolHash)
+    const newState = getState()
+    updateStakePoolIdentifier(newState, null, newState.poolRecommendation.recommendedPoolHash)
+  }
+
+  /* MULTIPLE ACCOUNTS */
+
+  const loadAccount = async (state: State, accountIndex: number) => {
+    loadingAction(state, 'Loading account')
+    await wallet.loadAccount(accountIndex)
+    const accountInfo = await wallet.accounts[accountIndex].getAccountInfo(state.validStakepools)
+    setState({
+      accounts: {
+        ...state.accounts,
+        [accountIndex]: accountInfo,
+      },
+    })
+  }
+
+  const setAccount = async (state: State, accountIndex: number) => {
+    if (!wallet.accounts[accountIndex]) {
```

Comment:
is this supposed to happen? Shouldn't the accounts be already loaded when setting them?